griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+import logging
+from schema import Schema, Literal, Optional
+from attr import define, field
+from griptape.artifacts import ErrorArtifact, InfoArtifact, ListArtifact, BlobArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+from googleapiclient.errors import HttpError
+from google.auth.exceptions import MalformedError
+from googleapiclient.discovery import Resource
+
+@define
+class GoogleDriveClient(BaseGoogleClient):
+    owner_email: str = field(kw_only=True)
+    LIST_FILES_SCOPES = ['https://www.googleapis.com/auth/drive.readonly']
+    UPLOAD_FILE_SCOPES = ['https://www.googleapis.com/auth/drive.file']
+    max_files: int = field(default=10, kw_only=True)
+
+    def _build_client(self, scopes: list[str]) -> Resource:
+        from google.oauth2 import service_account
+        from googleapiclient.discovery import build
+        credentials = service_account.Credentials.from_service_account_info(
+            self.service_account_credentials, scopes=scopes
+        )
+        delegated_credentials = credentials.with_subject(self.owner_email)
+        service = build('drive', 'v3', credentials=delegated_credentials)
+        return service
+
+    def list_files(self, params: dict) -> ListArtifact | ErrorArtifact:
+        values = params["values"]
+
+        try:
+            service = self._build_client(self.LIST_FILES_SCOPES)
+
+            folder_id = None
+            if "folder_path" in values and values["folder_path"]:
```

Comment:
You don't have to check for the key since we do that during LLM input validation.

Reply:
Done!

Reply:
If this is not `Optional` anymore, you don't have to do the check.

As a side note: if it were optional then a much better way to check for `None` would be `values.get("folder_path")` which doesn't throw an exception on `None` unlike `values["folder_path"]`.